### PR TITLE
Improve README examples

### PR DESCRIPTION
**Pull request by dlsniper**
This PR adds some improved examples of how to use XRay from Go.

### CODE REVIEW

**Comment by dlsniper** [comment]
I've used @alexbilbie's example from #1 to add the example of using a custom segment / subsegment as I was basically too lazy to type it on my own. Not sure how this works out for copyright, please advise.

**Comment by dlsniper** [comment]
Once #7 lands, errors can be added as well.

**Comment by luluzhao** [comment]
Hi @dlsniper, thank you for your suggestions about refactoring README. I gonna start modify it based on your suggestions. I agree with #1 , however, for #2 , I gonna keep glide section since some of customer might use it and what is more, "go get -u github.com/aws/aws-xray-sdk-go/..." only install non-test packages and it customer want to run unit test. They still need to install dependencies for unit-test by themselves.

**Comment by dlsniper** [comment]
> I gonna keep glide section since some of customer might use it and what is more, "go get -u github.com/aws/aws-xray-sdk-go/..." only install non-test packages and it customer want to run unit test. They still need to install dependencies for unit-test by themselves.

You can add ` -t ` to ` go get ` for getting the testing dependencies. I understand that you prefer having them at a specific version, but this won't work too well for various dependency managers if the user runs ` glide install `. I agree that overall this is not a good solution but sadly that's the current state of package management in Go.

Feel free to send the changes in the branch, it's open to changes form maintainers.

File: README.md [hunk 1]
```diff
@@ -83,17 +85,40 @@ func criticalSection(ctx context.Context) {
 **HTTP Handler**
 
 ```go
-func main() {
-  http.Handle("/", xray.Handler(xray.NewFixedSegmentNamer("myApp"), http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+package main
```
Comment (luluzhao) [inline README.md]: From my opinion, I don't think it's a good idea to add library information into every example. Lots of code are just repeated.

Comment (dlsniper) [reply README.md]: The problem that I had (recently when I started to need to integrate x-ray again) is that I couldn't find these easy to do. Maybe a better way would be to have a few [testable examples](https://blog.golang.org/examples) instead? I'm happy to redo the PR if so.

**Comment by luluzhao** [comment]
@dlsniper , I am sorry we get back to you after a long time. We updated our README and took part of your suggestion. We decide to keep `glide install` section since some customer may use it. And also I think it's not a good idea to write down library in every example since lots of code are just repeated again and again.

File: README.md (hunk 1 above)
Comment (luluzhao) [reply README.md]: In README configuration section, there is a part to tell customer to import xray library. Have you take a look at that? Could you please tell me which example you were getting stuck into?

**Pull request closed without merging**